### DISCUSSION TITLE
How to use access bookmark in macOS?

### DISCUSSION THREAD

**Discussion by pzs7602** [AvaloniaUI/Avalonia#10084] [Q&A] [answered]
I want to publish my Avalonia desktop application in Apple's App Store, the app will run in sandbox. In native macOS application, bookmark data is required for sandboxed applications that need to access a media file outside the sandboxed environment. (https://developer.apple.com/documentation/professional_video_applications/fcpxml_reference/asset/media-rep/bookmark). But how to do in an Avalonia desktop application when it will be published to App Store? For instance, I want to open a file by using StorageProvider.OpenFilePickerAsync method?

**Comment by maxkatz6** [comment, marked as answer]
It's not yet implemented but is planned. 
In general, yes, you need to use StorageProvider to open file, save bookmark and open file from the bookmark on next execution.

There was this PR https://github.com/AvaloniaUI/Avalonia/pull/6540 which needs to be redone using new API of bookmarks added in later PR https://github.com/AvaloniaUI/Avalonia/pull/8303.

**Reply by pzs7602** [reply]
Thanks for the reply. Will this be implemented in v11.0 release?

**Reply by maxkatz6** [reply]
Not sure if I will have time to pick it up before the release, as next couple of months for me will be quite busy. 
If anybody else is interested in doing it - I will try to help.
